oci/spec_opts_test.go

```diff
@@ -635,53 +634,75 @@ func TestDropCaps(t *testing.T) {
 
 func TestDevShmSize(t *testing.T) {
 	t.Parallel()
-	var (
-		s   Spec
-		c   = containers.Container{ID: t.Name()}
-		ctx = namespaces.WithNamespace(context.Background(), "test")
-	)
 
-	err := populateDefaultUnixSpec(ctx, &s, c.ID)
-	if err != nil {
-		t.Fatal(err)
+	ss := []Spec{
+		{
+			Mounts: []specs.Mount{
+				{
+					Destination: "/dev/shm",
+					Type:        "tmpfs",
+					Source:      "shm",
+					Options:     []string{"nosuid", "noexec", "nodev", "mode=1777"},
+				},
+			},
+		},
+		{
+			Mounts: []specs.Mount{
+				{
+					Destination: "/test/shm",
+					Type:        "tmpfs",
+					Source:      "shm",
+					Options:     []string{"nosuid", "noexec", "nodev", "mode=1777", "size=65536k"},
+				},
+				{
+					Destination: "/dev/shm",
+					Type:        "tmpfs",
+					Source:      "shm",
+					Options:     []string{"nosuid", "noexec", "nodev", "mode=1777", "size=65536k", "size=131072k"},
```

Comment:
could you check the options in case? since you de-duplicate options..

Reply:
Only the size option is de-deplicated, because this function only operates on the size option and does not modify any other options.

Do you want to check if other options have been modified?

Reply:
The size option is de-duplicated to ensure that the correct size is set, as the last size option is used when mounting.
We can also use other methods, such as setting all size options
```go
func WithDevShmSize(kb int64) SpecOpts {
	return func(ctx context.Context, _ Client, _ *containers.Container, s *Spec) error {
		for i, m := range s.Mounts {
			if m.Destination == "/dev/shm" && m.Source == "shm" && m.Type == "tmpfs" {
				for i := 0; i < len(m.Options); i++ {
					if strings.HasPrefix(m.Options[i], "size=") {
						m.Options =  fmt.Sprintf("size=%dk", kb)
					}
				}
				return nil
			}
		}
		return ErrNoShmMount
	}
}
```
What do you think is the better way.

Reply:
I think we can add validation about options, like:

```
input: []string{"nosuid", "noexec", "size=65k", "nodev", "mode=1777", "size=65536k"},
expected: []string{"nosuid", "noexec", "nodev", "mode=1777", "size=1024k"},
```

Reply:
Check the size options in the [`getShmSize`](https://github.com/containerd/containerd/pull/5063/files#diff-ab0953a9b7d47786cfd9038c463ffbc79a6df7b299429072b3a5e14a735ec170R695)
Is this okay?

Reply:
OK. Please squash commits into one. Thanks